src/main/java/Server/MopeLSPServerLauncher.java

```diff
@@ -29,13 +29,22 @@
     private static Logger logger = LoggerFactory.getLogger(MopeLSPServerLauncher.class);
     private static ConfigObject configObject;
 
+    /**
+     * <p>starts a new server and server-socket</p>
+     * @throws IOException in case of an I/O error
+     */
     public MopeLSPServerLauncher() throws IOException {
         configObject = new ConfigObject();
         readConfig();
         server = new MopeLSPServer(configObject);
         serverSocket = new ServerSocket(configObject.port);
     }
 
+    /**
+     * <p>launches the server and accepts the connection of the client</p>
+     * @throws ExecutionException in case of attempting to retrieve the result of a task that aborted by throwing an exception
+     * @throws InterruptedException in case of a thread gets interrupted
```

Comment:
Here we are in the launcher, which controls the interrupt policy. I do not think the main method of the launcher should throw any `InterruptedExceptions`. This is the point where you can convert them to `AssertionError`s.